src/components/SarStakingWidget/Stake/index.tsx

```diff
@@ -0,0 +1,200 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { Drawer } from 'src/components';
+import { Box } from 'src/components/Box';
+import { Button } from 'src/components/Button';
+import { Text } from 'src/components/Text';
+import { TextInput } from 'src/components/TextInput';
+import { ZERO_ADDRESS } from 'src/constants';
+import { PNG } from 'src/constants/tokens';
+import { useChainId, usePangolinWeb3 } from 'src/hooks';
+import { ApprovalState } from 'src/hooks/useApproveCallback';
+import { useWalletModalToggle } from 'src/state/papplication/hooks';
+import { useDerivativeSarStake, useSarStakeInfo } from 'src/state/psarstake/hooks';
+import { useTokenBalances } from 'src/state/pwallet/hooks';
+import DrawerContent from './DrawerContent';
+import { Buttons, Root } from './styleds';
+
+export default function Stake() {
+  const [openDrawer, setOpenDrawer] = useState(false);
+
+  const chainId = useChainId();
+  const { account } = usePangolinWeb3();
+
+  const png = PNG[chainId];
+  const tokensBalances = useTokenBalances(account ?? '', [png]);
+  const userPngBalance = tokensBalances[png.address];
```

Comment:
you can just use `useTokenBalance` instead of `useTokenBalances` because you just want to fetch one token balance i.e. png
Please do this change everywhere you used `useTokenBalances`